app/src/Vues/Register.jsx

```diff
@@ -0,0 +1,201 @@
+    const [prenomError, setprenomError] = useState('');
+    const [nomError, setnomError] = useState('');
+    const [passwordError, setPasswordError] = useState('');
+    const [confirmPasswordError, setConfirmPasswordError] = useState('');
+    const [loginError, setLoginError] = useState('');
+
+    const [generalError, setGeneralError] = useState("");
+    const [success, setSuccess] = useState(false);
+
+    const handleSignUp = async () => {
+                const userData = {
+                    id : 0,
+                    login: login,
+                    firstname: prenom,
+                    lastname: nom,
+                    mail: email,
+                };
+
+                setSuccess(true)
+
+                try {
+
+                    const userModel = new UserModel(userData);
+                    userModel.password = password
+                    userModel.password_confirm = confirmPassword
+
+                    const response = await userModel.createUser();
+
+                    if (response instanceof ErrorResponse) {
+                        setSuccess(false)
+                        if (response.errorCode() === 401) {
+                            setGeneralError("Identifiants incorrects");
+                        } else {
+                            setGeneralError(`Une erreur est survenue: ${response.errorMessage()}`);
+                        }
+                        return;
+                    }
+
+                } catch (err) {
+                    setSuccess(false)
+                    const errorMessage = err instanceof Error ? err.message : "Une erreur est survenue";
+                    setGeneralError(`Une erreur est survenue: ${errorMessage}`);
+                }
+    };
+
+    const handleMailType = (e) => {
+
+        const re = new RegExp("^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$")
+        if (re.test(e.target.value)){
+            setEmailError("")
+        }else {
+            setEmailError("Veuillez entrer une adresse email valide.")
+        }
+        //console.log(emailError)
+        setEmail(e.target.value)
+    };
+
+    const handlePrenom = (e) => {
+
+        if (e.target.value.length >= 2){
+            setprenomError("")
+
+        }else {
+            setprenomError("Veuillez entrer un prénom valide, d'au moins 2 caractères.")
+        }
+        setPrenom(e.target.value)
+    };
+
+    const handleNom = (e) => {
+
+        if (e.target.value.length >= 2){
+            setnomError("")
+
+        }else {
+            setnomError("Veuillez entrer un nom valide, d'au moins 2 caractères.")
+        }
+        setNom(e.target.value)
+    };
+
+    const handleLogin = (e) => {
+
+        if (e.target.value.length >= 2){
+            setLoginError("")
+
+        }else {
+            setLoginError("Veuillez entrer un login valide, d'au moins 2 caractères.")
+        }
+        setLogin(e.target.value)
+    };
+
+    const handleMotDePasse = (e) => {
```

Comment:
Maybe move this into a utils file.

Reply:
en vrai, hormis dans le Form on va pas les utiliser beaucoup donc je pense les laisser dans le form en vrai